File: src/demo/scenario.js
import { mkdtempSync, rmSync } from 'node:fs';
import { tmpdir } from 'node:os';
import { join } from 'node:path';
import { createEngine, formatState } from './helpers.js';

const rootDir = mkdtempSync(join(tmpdir(), 'aleph-scenario-'));
const engine = createEngine(rootDir);

function printTurn(label, result) {
  console.log(`\n=== ${label} ===`);
  console.log(`Active persona: ${result.activePersonaName} (${result.activePersonaId})`);
  if (result.switchDecision?.approved) {
    console.log(`Switch: ${result.switchDecision.explanation}`);
    console.log(result.switchDecision.handoffSummary);
  }
  console.log(`Reply: ${result.reply}`);
}

try {
  console.log('Aleph scripted scenario');
  console.log('-----------------------');
  console.log(formatState(engine.inspectState()));

  const first = engine.processUserTurn(
    '我刚刚答应合作方今晚之前给他回一版方案，但我现在开始慌了。',
  );
  printTurn('Turn 1', first);

  const second = engine.processUserTurn(
    '你来拍板并接管吧，我们必须推进，不然这个承诺会失控。',
  );
  printTurn('Turn 2', second);

  const third = engine.processUserTurn(
    '另外我刚刚说重了话，关系有点僵，这件事不会因为换人就消失。',
  );
  printTurn('Turn 3', third);

  console.log('\n=== Final state ===');
  console.log(formatState(engine.inspectState()));
} finally {
  engine.store.close();
  rmSync(rootDir, { recursive: true, force: true });
}

File: aleph/demo/scenario.py
from __future__ import annotations

import shutil
from pathlib import Path

from aleph.demo.helpers import create_engine, format_state


def _print_stream(events: list[dict]) -> None:
    for event in events:
        payload = event["payload"]
        if event["event_kind"] == "delta":
            print(f"{event['source']}> {payload['text']}")
        elif event["event_kind"] == "handoff":
            print(f"[handoff] {payload['explanation']}")
            print(payload["handoff_summary"])
        elif event["event_kind"] == "status":
            print(f"[status] {payload['message']}")
        elif event["event_kind"] == "tool_event":
            print(f"[tool] {payload['tool_id']} -> {payload['summary']}")


def main() -> None:
    workspace_tmp = Path.cwd() / ".aleph-tmp" / "scenario-workdir"
    shutil.rmtree(workspace_tmp, ignore_errors=True)
    workspace_tmp.mkdir(parents=True, exist_ok=True)
    engine = create_engine(workspace_tmp)
    try:
        print("Aleph scripted scenario")
        print("-----------------------")
        print(format_state(engine.inspect_state()))

        turns = [
            "请记住，我答应今晚之前给合作方一个版本。",
            "你来拍板并且推进这件事。",
            "我刚才说重了，关系有点僵。",
        ]
        for index, turn in enumerate(turns, start=1):
            print(f"\n=== Turn {index} ===")
            result = engine.process_user_turn(turn)
            _print_stream(result["stream"])

        print("\n=== Final state ===")
        print(format_state(engine.inspect_state()))
    finally:
        engine.store.close()
        shutil.rmtree(workspace_tmp, ignore_errors=True)


if __name__ == "__main__":
    main()
